Core/include/Acts/Propagator/Navigator.hpp

```diff
@@ -836,7 +832,7 @@ class Navigator {
       NavigationOptions<Surface> navOpts;
       // Exclude the current surface in case it's a boundary
       navOpts.startObject = state.navigation.currentSurface;
-      navOpts.nearLimit = stepper.overstepLimit(state.stepping);
+      navOpts.nearLimit = state.options.surfaceTolerance;
```

Comment:
Overstepping is now always resolved by the stepper by trying to reach the surface no matter how far we are overstepping. The idea of using the `surfaceTolerance` here was to only look for new intersections ahead of us since anything in range of `surfaceTolerance` would be considered to be "on surface".